datatypes/src/collections/data_types.rs

```diff
@@ -9,3 +12,37 @@ pub enum VectorDataType {
     MultiLineString,
     MultiPolygon,
 }
+
+#[derive(Debug)]
+pub enum TypedFeatureCollection {
+    Data(DataCollection),
+    MultiPoint(MultiPointCollection),
+    MultiLineString(MultiLineStringCollection),
+    MultiPolygon(MultiPolygonCollection),
+}
+
+impl TypedFeatureCollection {
+    pub fn vector_data_type(&self) -> VectorDataType {
+        match self {
+            TypedFeatureCollection::Data(_) => VectorDataType::Data,
+            TypedFeatureCollection::MultiPoint(_) => VectorDataType::MultiPoint,
+            TypedFeatureCollection::MultiLineString(_) => VectorDataType::MultiLineString,
+            TypedFeatureCollection::MultiPolygon(_) => VectorDataType::MultiPolygon,
+        }
+    }
+
+    // TODO: other types
```

Comment:
Mabye resolve this by adding the other two.

Reply:
We could also add a trait for this and implement it for the Types.

Reply:
0c148c7dc15557c77a7dd6d7f9cd11c43545bfcd